sdk/include/reactos/imm32_undoc.h

```diff
@@ -7,9 +7,23 @@
 
 #pragma once
 
-#define KBDLAYOUT_MASK 0xF000
-#define KBDLAYOUT_IME 0xE000
-#define IS_IME_KBDLAYOUT(hKL) ((HIWORD(hKL) & KBDLAYOUT_MASK) == KBDLAYOUT_IME)
+/* unconfirmed */
+typedef struct tagCLIENTIMC
+{
+    HIMC hImc;                  // offset 0x0
+    LONG cLockObj;              // offset 0x4
+    DWORD dwFlags;              // offset 0x8
+    DWORD unknown;              // offset 0xc
+    RTL_CRITICAL_SECTION cs;    // offset 0x10
+    DWORD unknown2;             // offset 0x28
+    DWORD unknown3;             // offset 0x2c
+    BOOL bUnknown4;             // offset 0x30
```

Comment:
I would strongly suggest to remove these offsets, as otherwise this looks suspicious (if you understand what I mean...)

Reply:
Removed.